tests/de.uka.ipd.sdq.simucomframework.tests/src/de/uka/ipd/sdq/simucomframework/tests/SimulatedResourceContainerTests.java

```diff
@@ -0,0 +1,202 @@
+import de.uka.ipd.sdq.simucomframework.ResourceRegistry;
+import de.uka.ipd.sdq.simucomframework.SimuComConfig;
+import de.uka.ipd.sdq.simucomframework.SimuComSimProcess;
+import de.uka.ipd.sdq.simucomframework.exceptions.ResourceContainerIsMissingRequiredResourceType;
+import de.uka.ipd.sdq.simucomframework.model.SimuComModel;
+import de.uka.ipd.sdq.simucomframework.resources.SchedulingStrategy;
+import de.uka.ipd.sdq.simucomframework.resources.SimulatedResourceContainer;
+import de.uka.ipd.sdq.simucomframework.simucomstatus.SimucomstatusFactory;
+import de.uka.ipd.sdq.simulation.abstractsimengine.IEntity;
+import de.uka.ipd.sdq.simulation.abstractsimengine.ISimEngineFactory;
+
+class SimulatedResourceContainerTests {
+
+	// Mocks
+	SimuComModel mockModel;
+	ResourceRegistry resourceRegistry;
+	ISimEngineFactory simEngineFactory;
+	ISchedulingFactory schedulingFactory;
+	SimuComConfig simuComConfig;
+
+	@BeforeEach
+	void createMockModel() {
+
+		mockModel = mock(SimuComModel.class);
+
+		simEngineFactory = mock(ISimEngineFactory.class);
+		when(mockModel.getSimEngineFactory()).thenReturn(simEngineFactory);
+
+		resourceRegistry = new ResourceRegistry(mockModel);
+		when(mockModel.getResourceRegistry()).thenReturn(resourceRegistry);
+		
+		var status = SimucomstatusFactory.eINSTANCE.createSimuComStatus();
+		status.setResourceStatus(SimucomstatusFactory.eINSTANCE.createSimulatedResources());
+		when(mockModel.getSimulationStatus()).thenReturn(status);
+		
+		schedulingFactory = mock(ISchedulingFactory.class);
+		when(mockModel.getSchedulingFactory()).thenReturn(schedulingFactory);
+		
+		simuComConfig = mock(SimuComConfig.class);
+		when(simuComConfig.getSimulateFailures()).thenReturn(false);
+		when(mockModel.getConfiguration()).thenReturn(simuComConfig);
+		
+	}
+
+	@Test
+	void testBasicNestedResourceContainer() {
+
+		// Given a container with a single child container
+		String parentContainerId = "container0";
+		SimulatedResourceContainer parentContainer = new SimulatedResourceContainer(mockModel, parentContainerId);
+		resourceRegistry.addResourceContainer(parentContainer);
+
+		String childContainerId = "container1";
+		SimulatedResourceContainer childContainer = new SimulatedResourceContainer(mockModel, childContainerId);
+		resourceRegistry.addResourceContainer(childContainer);
+		childContainer.setParentResourceContainer(parentContainerId);
+		parentContainer.addNestedResourceContainer(childContainerId);
+
+		var nested = parentContainer.getNestedResourceContainers();
+		assertEquals(1, nested.size(), "we should have one nested container");
+
+		var foundContainer = nested.get(0);
+		assertEquals(childContainerId, foundContainer.getResourceContainerID(), "nested container should have our Id");
+		assertEquals(parentContainer, foundContainer.getParentResourceContainer(),
+				"child should have our parent as container");
+
+	}
+
+	@Test
+	void testNestedResourceContainerCalls() {
```

Comment:
What does this test test? All calls for loading a resource are done on the parent container. I was wondering, what effect the child container actually has in this test case. Intuitively, I would think that the calls should be done on the child container to verify that the parent container is called.

Reply:
should be fixed, was just too much mockery :)